### PR TITLE
[wasm] renames and cleanup before modularization

### PR DESCRIPTION
**Pull request by pavelsavara**
- rename main javaScript files to `main.js` and `test-main.js`
- sample and test script cleanup
- rename set_exit_code method
- rewrite test start as async method
- improve script loading via script element on page in test
- use BINDING.bind_static_method instead of INTERNAL where possible
- better .d.ts exports
- formatted html files
- moved dotnet IFFE out of emcc Closure compiled section and re-enabled terser() and ES2018 target.

### CODE REVIEW

**Comment by kg** [comment]
Happy to finally rename runtime to something reasonable

**Comment by pavelsavara** [comment]
This passed CI tests with full matrix. I also tested debugger unit tests and debugger in Blazor app manually.

File: src/mono/wasm/runtime/export-types.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { DotNetExports } from "./exports";
+import { EmscriptenModuleConfig } from "./types";
+
+// this is the only public export from the dotnet.js module
+declare function createDotnetRuntime(moduleFactory: (api: DotNetExports) => EmscriptenModuleConfig): Promise<DotNetExports>;
```
Comment (kg) [inline src/mono/wasm/runtime/export-types.ts:5]: Maybe DotNetExports is the wrong name for this if 'createDotnetRuntime' is the only public export from dotnet.js? If the parameter name is 'api' maybe it's DotNetPublicAPI or something

File: src/mono/wasm/runtime/exports.ts [hunk 2]
```diff
@@ -115,23 +115,48 @@ export const BINDING: BINDING = <any>{
 // this is executed early during load of emscripten runtime
 // it exports methods to global objects MONO, BINDING and Module in backward compatible way
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-function export_to_emscripten(dotnet: any, mono: any, binding: any, internal: any, module: any): void {
+function exportAPI(mono: any, binding: any, internal: any, module: any, isGlobal: boolean, isNode: boolean, isShell: boolean, isWeb: boolean, glocateFile: Function): void {
```
Comment (kg) [inline src/mono/wasm/runtime/exports.ts:118]: Can we make all these bools and glocateFile live in an object with keys?

File: src/mono/wasm/runtime/startup.ts [hunk 3]
```diff
@@ -223,6 +238,20 @@ function _finalize_startup(args: MonoConfig, ctx: MonoInitContext) {
...
+    const argsAny: any = args;
+    if (argsAny.loaded_cb) {
+        try {
+            argsAny.loaded_cb();
+        }
+        catch (err: any) {
+            Module.printErr("MONO_WASM: loaded_cb () failed: " + err);
+            Module.printErr("MONO_WASM: Stacktrace: \n");
+            Module.printErr(err.stack);
+            throw err;
```
Comment (kg) [inline src/mono/wasm/runtime/startup.ts:251]: shouldn't we reject here?

File: src/mono/wasm/test-main.js [hunk 4]
```diff
@@ -101,18 +101,12 @@ if (typeof globalThis.performance === 'undefined') {
...
     config: null,
     configSrc: "./mono-config.json",
-    print: console.log,
-    printErr: console.error,
-    onConfigLoaded: function () {
+    onConfigLoaded: () => {
         if (!Module.config) {
-            console.error("Could not find ./mono-config.json. Cancelling run");
-            fail_exec(1);
+            set_exit_code(1, "Could not find ./mono-config.json. Cancelling run");
```
Comment (kg) [inline src/mono/wasm/test-main.js:109]: while i think the extra code below running is harmless, we should probably 'return' here after setting the exit code anyway

**Review by kg — APPROVED** [review]

File: src/mono/wasm/test-main.js [hunk 5]
```diff
@@ -333,7 +326,17 @@ async function loadDotnet(file) {
             const script = document.createElement("script");
             script.src = file;
             document.head.appendChild(script);
-            return globalThis.Module;
+            let timeout = 100;
+            // bysy spin waiting for script to load into global namespace
+            while (timeout > 0) {
```
[the remark this answers is not in the document]
Comment (kg) [reply src/mono/wasm/test-main.js:331]: if the script being loaded has to do some initialization work, the 'load' event might be too early

[the remark this answers is not in the document]
Comment (maraf) [reply src/mono/wasm/test-main.js:331]: Ook 👍

File: src/mono/wasm/runtime/exports.ts (hunk 2 above)
Comment (pavelsavara) [reply src/mono/wasm/runtime/exports.ts:118]: They are not bools, `mono` is seed object what would become `MONO` later. `isNode`, `locateFile` etc is import from emscripten which is not expoted on `Module` object. I will rename it to make it clearer.

File: src/mono/wasm/runtime/exports.ts [hunk 6]
```diff
@@ -314,4 +337,9 @@ export interface BINDING {
     bind_static_method: typeof mono_bind_static_method,
     call_assembly_entry_point: typeof mono_call_assembly_entry_point,
     unbox_mono_obj: typeof unbox_mono_obj
+}
+export interface DotNetExports {
+    MONO: MONO,
+    BINDING: BINDING,
+    Module: any
 }
```
[the remark this answers is not in the document]
Comment (pavelsavara) [reply src/mono/wasm/runtime/exports.ts:345]: Yes, this is only type export, not actual API instance. I don't want to document that we have INTERNAL in the `.d.ts` file

File: src/mono/wasm/test-main.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (pavelsavara) [reply src/mono/wasm/test-main.js:331]: with ES6, this will do away and we will only `import`.

**Comment by thaystg** [comment]
Debugger tests -> passing.
Debug a Blazor App -> working.

**Pull request merged**